src/rules/export.js

```diff
@@ -34,6 +47,7 @@ module.exports = {
 
   create: function (context) {
     const namespace = new Map([[rootProgram, new Map()]])
+    const isTypescriptFile = /\.ts|\.tsx$/.test(context.getFilename())
```

Comment:
this seems like something that would need to be determined based on parser settings, not hardcoded extensions - also, what about flow?

Reply:
1. Will it suffice to do something like `\/@typescript-eslint\/parser\/.test(context.parserPath)`?(`context.parserPath` is like `/home/some-user/some-project/node_modules/@typescript-eslint/parser/dist/parser.js`) 2. It seems that flow does not accept function overload like the case here.

Reply:
that would force people to install the typescript parser even if they weren’t using typescript, i think - ideally we’d find a more reliable mechanism than “its typescript” to handle this case.

Reply:
Just find out find that these function nodes without body has node type `TSDeclareFunction`. Therefore I change the bail out logic to `some nodes is TSDeclareFunction && every nodes is TSDeclareFunction or FunctionDeclaration`.